nucliadb/nucliadb/writer/back_pressure.py

```diff
@@ -434,17 +436,18 @@ def check_ingest_behind(ingest_pending: int):
 
     if ingest_pending > max_pending:
         try_after = estimate_try_after(
-            rate=settings.ingest_rate, pending=ingest_pending
+            rate=settings.ingest_rate, pending=ingest_pending, max_pending=max_pending
         )
         data = BackPressureData(type="ingest", try_after=try_after)
         raise BackPressureException(data)
 
 
-def estimate_try_after(rate: float, pending: int) -> datetime:
+def estimate_try_after(rate: float, pending: int, max_pending: int) -> datetime:
     """
     This function estimates the time to try again based on the rate and the number of pending messages.
     """
-    delta_seconds = pending / rate
+    # We estimate the time to try again when the pending messages are half of the max_pending
+    delta_seconds = (pending - max_pending / 2) / rate
```

Comment:
You're relying on `pending` being greater than `max_pending / 2` but the conditional is done in another function